python/ray/data/grouped_dataset.py

```diff
@@ -249,6 +252,10 @@ def group_fn(batch):
             for end in boundaries:
                 group = block_accessor.slice(start, end, False)
                 applied = fn(group)
+                if applied is None or not isinstance(applied, type(batch)):
+                    logger.error("Expecting output type %s, but got %s.",
```

Comment:
```suggestion
                    raise ValueError("Expecting output type %s, but got %s.",
```

Btw, is the type check quite correct? I think block builder can take multiple possible types. Maybe the check should be moved into `add_block()`.

Reply:
It should be better to raise error here since it has the right context (i.e. UDF should return the same type as the input data). Inside add_block() we don't know why exactly there is a different type added.